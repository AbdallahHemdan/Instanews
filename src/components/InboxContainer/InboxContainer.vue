<template>
  <div class="inbox">
    <div class="inbox__logo">
      <img src="./../../assets/svgs/message.svg" alt="message logo" />
    </div>

    <div class="inbox__info">
      <div class="inbox__main-title">
        You Messages
      </div>

      <div class="inbox__sub-title">
        Send private photos and messages to a friend or group.
      </div>

      <button class="inbox__sent-btn">Send Message</button>
    </div>
  </div>
</template>

<script>
export default {
  name: 'InboxContainer',
};
</script>

<style lang="scss" scoped>
.inbox {
  height: 87vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  &__logo {
    width: 96px;
    height: 96px;
  }

  &__info {
    margin-top: 32px;
    text-align: center;
  }

  &__main-title {
    font-size: 22px;
    line-height: 26px;
    margin-bottom: 0.5rem;
  }

  &__sub-title {
    color: $dark-gray;
    font-size: $font-size-2x;
  }

  &__sent-btn {
    color: $white;
    background-color: $main-color;
    border: 1px solid transparent;
    border-radius: 4px;
    font-weight: 600;
    margin-top: 2rem;
    padding: 4px 10px;
  }
}
</style>
