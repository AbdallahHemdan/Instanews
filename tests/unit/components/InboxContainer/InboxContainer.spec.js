import { shallowMount } from '@vue/test-utils';
import InboxContainer from './../../../../src/components/InboxContainer/InboxContainer';

describe('InboxContainer.vue', () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(InboxContainer);
  });

  describe('Test inner text of elements', () => {
    it('should render messages logo correctly', () => {
      expect(
        cmp
          .find('.inbox__logo')
          .find('img')
          .attributes('src'),
      ).toBe('./../../assets/svgs/message.svg');
    });

    it('should render main title correctly', () => {
      expect(cmp.find('.inbox__main-title').text()).toBe('You Messages');
    });

    it('should render subtitle correctly', () => {
      expect(cmp.find('.inbox__sub-title').text()).toBe(
        'Send private photos and messages to a friend or group.',
      );
    });

    it('should render inner text of sent message correctly', () => {
      expect(cmp.find('.inbox__sent-btn').text()).toBe('Send Message');
    });
  });

  describe('Snapshot testing', () => {
    it('should pass snapshot testing', () => {
      expect(cmp.element).toMatchSnapshot();
    });
  });
});
