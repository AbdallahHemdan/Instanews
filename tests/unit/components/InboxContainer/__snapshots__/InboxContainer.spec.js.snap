// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InboxContainer.vue Snapshot testing should pass snapshot testing 1`] = `
<div
  class="inbox"
>
  <div
    class="inbox__logo"
  >
    <img
      alt="message logo"
      src="./../../assets/svgs/message.svg"
    />
  </div>
   
  <div
    class="inbox__info"
  >
    <div
      class="inbox__main-title"
    >
      
      You Messages
    
    </div>
     
    <div
      class="inbox__sub-title"
    >
      
      Send private photos and messages to a friend or group.
    
    </div>
     
    <button
      class="inbox__sent-btn"
    >
      Send Message
    </button>
  </div>
</div>
`;
